src/renderer/services/db/abstract.ts

```diff
@@ -0,0 +1,7 @@
+import { object } from 'prop-types';
+
+export interface CGSerialization<T> {
+    serialize(value: any, type: T): Buffer;
+    deserialize(value: Buffer, type: T): any;
```

Comment:
I will answer you question here. We haven't really solved this one in ssz library as generics in typescript are kinda limited. So we can leave `any` here. 

I would only suggest for deserialize method to be
`deserialize<R>(value: Buffer, type: T): R`
as it's much easier to set type than doing `as Something` everywhere

Reply:
Also you can hardcode AnySSZType as type in CGSerializer (you don't need generics) because if we change serialization we will write adapter to convert AnySSZType to required schema rather than writing type descriptors for every type